tgui/packages/tgui/interfaces/ChemHeater.js

```diff
@@ -0,0 +1,123 @@
+              />
+            </Stack.Item>
+            <Stack.Item grow basis={0} overflow="hidden">
+              <Box
+                className="ChemHeater__TemperatureNumber"
+                nowrap
+                p={1}
+                fontSize={1.5}
+                color={getTemperatureColor(targetTemperature)}
+                backgroundColor="black">
+                <Box fontSize={1}>Target</Box>
+                <Icon name={getTemperatureIcon(targetTemperature)} pr={0.5} />
+                <AnimatedNumber value={targetTemperature} /> K
+              </Box>
+            </Stack.Item>
+            <Stack.Item basis={9.6} align="center">
+              <Button
+                icon="power-off"
+                disabled={!containerData?.totalVolume}
+                color={isActive ? "red" : "green"}
+                fluid
+                height="100%"
+                fontSize={1.25}
+                textAlign="center"
+                onClick={() => act(isActive ? 'stop' : 'start')}>
+                {isActive ? "Deactivate" : "Activate"}
+              </Button>
+            </Stack.Item>
+          </Stack>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ChemDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const { active = false, targetTemperature = freezeTemperature } = props;
+  const container = props.container ?? NoContainer;
+  const working = active && !container.fake;
+  const { temperature, totalVolume } = container;
+
+  return (
+    <SectionWithTitleOverflow
+      title={
+        <Box as="span"
+          style={{
+            "text-transform": "capitalize",
+          }}>
+          {container.name}
+        </Box>
+      }
+      buttons={
+        <Button
+          icon="eject"
+          disabled={!props.container}
+          onClick={() => act('eject')}>
+          Eject
+        </Button>
+      }>
+      <ReagentGraph container={container} />
+      <ReagentList container={container} />
+      <Box className={"ChemHeater__TemperatureBox " + (working && `ChemHeater__TemperatureBox__${getTemperatureChangeName(temperature, targetTemperature)}`)}>
+        {!totalVolume || (
+          <Box
+            fontSize={2}
+            color={getTemperatureColor(temperature)}
+            className={"TemperatureNumber"}>
+            <Icon name="long-arrow-alt-down"
+              className={"ChemHeater__TemperatureArrow " + (working && `ChemHeater__TemperatureArrow__${getTemperatureChangeName(temperature, targetTemperature)}`)}
+              pt="2px"
+              pr={0.25}
+              style={{
+                transform: active ? `scaleY(${Math.sign(temperature - targetTemperature)})` : "scaleY(0)",
+              }}
+            />
+            <Icon name={getTemperatureIcon(temperature)} pr={0.5} />
+            <AnimatedNumber value={temperature} /> K
+          </Box>
+        )}
+      </Box>
+      {!!props.container || (
```

Comment:
Assuming intended logic here is "show the thing below if you don't have a container", would prefer written as:
```suggestion
      {!props.container && (
```
If that's not the logic you're going for then I'm not sure what this is doing.